File: ios/Runner/Bridges/TryKmpHost.swift
import TryKMP

internal class TryKmpHost: TryKmpHostApi {
    private let model = TryKmp()
    
       
    func searchGitHubRepo(query: String, completion: @escaping (Result<GitHubRepoDto, any Error>) -> Void) {
        model.searchGitHubRepo(query: query) { result, error in
            if let result = result {
                let mapped = GitHubRepoDto(
                    totalCount: Int64(result.totalCount),
                    incompleteResults: result.incompleteResults,
                    items: result.items.map {
                        GitHubRepoItemDto(
                            fullName: $0.fullName,
                            description: $0.description,
                            url: $0.url,
                            updatedAt: $0.updatedAt.ISO8601Format(),
                            language: $0.language
                        )
                    }
                )
                completion(.success(mapped))
            }
            completion(.failure(error ?? PigeonError(code: "", message: nil, details: nil)))
        }
    }
    
    func time() throws -> String {
        return model.time(suffix: "")
    }
}

File: android/app/src/main/kotlin/com/example/try_flutter/bridges/TryKmpHost.kt
package com.example.try_flutter.bridges

import io.github.tshion.trykmp.TryKmp
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

internal class TryKmpHost(
    private val scope: CoroutineScope,
) : TryKmpHostApi {

    private val model = TryKmp()


    override fun time(): String {
        return model.time()
    }

    override fun searchGitHubRepo(
        query: String,
        callback: (Result<GitHubRepoDto>) -> Unit
    ) {
        scope.launch {
            withContext(Dispatchers.Default) {
                try {
                    val result = model.searchGitHubRepo(query)
                    result.let { native ->
                        GitHubRepoDto(
                            native.totalCount.toLong(),
                            native.incompleteResults,
                            native.items.map {
                                GitHubRepoItemDto(
                                    it.fullName,
                                    it.description,
                                    it.url,
                                    it.updatedAt.toString(),
                                    it.language,
                                )
                            },
                        )
                    }.also { callback(Result.success(it)) }
                } catch (e: Exception) {
                    callback(Result.failure(e))
                }
            }
        }
    }
}
